fabric-structure-api-v1/src/main/java/net/fabricmc/fabric/api/structure/v1/FabricStructureBuilder.java

```diff
@@ -0,0 +1,201 @@
+	 *                structure must have an unique {@code salt} in order to spawn in different places.
+	 *
+	 * @see #defaultConfig(StructureConfig)
+	 */
+	public FabricStructureBuilder<FC, S> defaultConfig(int spacing, int separation, int salt) {
+		return defaultConfig(new StructureConfig(spacing, separation, salt));
+	}
+
+	/**
+	 * Sets the structure configuration which spawns in superflat worlds. If unset, this structure will not spawn in
+	 * superflat worlds.
+	 *
+	 * @see #superflatFeature(FeatureConfig)
+	 */
+	public FabricStructureBuilder<FC, S> superflatFeature(ConfiguredStructureFeature<FC, ? extends StructureFeature<FC>> superflatFeature) {
+		Objects.requireNonNull(superflatFeature, "superflatFeature must not be null");
+		this.superflatFeature = superflatFeature;
+		return this;
+	}
+
+	/**
+	 * Sets the structure configuration which spawns in superflat worlds. If unset, this structure will not spawn in
+	 * superflat worlds.
+	 *
+	 * @see #superflatFeature(ConfiguredStructureFeature)
+	 */
+	public FabricStructureBuilder<FC, S> superflatFeature(FC config) {
+		return superflatFeature(structure.configure(config));
+	}
+
+	/**
+	 * Causes structure pieces of this structure to adjust the surface of the world to fit them, so that they don't
+	 * stick out of or into the ground.
+	 */
+	public FabricStructureBuilder<FC, S> adjustsSurface() {
+		this.adjustsSurface = true;
+		return this;
+	}
+
+	/**
+	 * Registers this structure and applies the other changes from the {@linkplain FabricStructureBuilder}.
+	 */
+	public S register() {
+		Objects.requireNonNull(step, "Structure \"" + id + "\" is missing a generation step");
+		Objects.requireNonNull(defaultConfig, "Structure \"" + id + "\" is missing a default config");
+
+		// Ensure StructuresConfig class is initialized, so the assertion in its static {} block doesn't fail
+		StructuresConfig.DEFAULT_STRUCTURES.size();
+
+		StructureFeatureAccessor.callRegister(id.toString(), structure, step);
+
+		if (!id.toString().equals(structure.getName())) {
+			// mods should not be overriding getName, but if they do and it's incorrect, this gives an error
+			throw new IllegalStateException(String.format("Structure \"%s\" has mismatching name \"%s\". Structures should not override \"getName\".", id, structure.getName()));
+		}
+
+		StructuresConfigAccessor.setDefaultStructures(ImmutableMap.<StructureFeature<?>, StructureConfig>builder()
+				.putAll(StructuresConfig.DEFAULT_STRUCTURES)
+				.put(structure, defaultConfig)
+				.build());
+
+		if (superflatFeature != null) {
+			FlatChunkGeneratorConfigAccessor.getStructureToFeatures().put(structure, superflatFeature);
+		}
+
+		if (adjustsSurface) {
+			StructureFeatureAccessor.setSurfaceAdjustingStructures(ImmutableList.<StructureFeature<?>>builder()
+					.addAll(StructureFeature.field_24861)
+					.add(structure)
+					.build());
+		}
+
+		// update existing structure configs
```

Comment:
Feel free to ignore this one
```suggestion
		// update existing sets of structure configs
```